# HG changeset patch
# User jwe
# Date 1201976024 0
#      Sat Feb 02 18:13:44 2008 +0000
# Branch release-3-0-x
# Node ID 1b01b40886ebbb98eb55bb952a6fc077f21fe7d5
# Parent  bc19cf26ff0bf72127e718a9c7f0930c8e4df8a4
[3-0-0-branch @ 2008-02-02 18:13:44 by jwe]

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2008-01-30  Thomas Weber  <thomas.weber.mail@gmail.com>
+
+	* variables.cc (Fclear): Doc fix.
+
 2008-01-30  Thomas Weber  <thomas.weber.mail@gmail.com>
 
 	* pager.cc (Fmore): Doc fix.
 
 2008-01-28  John W. Eaton  <jwe@octave.org>
 
 	* oct-stream.cc (BEGIN_CHAR_CLASS_CONVERSION): Handle width properly.
 	(OCTAVE_SCAN) [__GNUG__ && ! CXX_ISO_COMPLIANT_LIBRARY]:
diff --git a/src/variables.cc b/src/variables.cc
--- a/src/variables.cc
+++ b/src/variables.cc
@@ -2387,17 +2387,17 @@ clear_variable (const std::string& nm)
 bool
 clear_symbol (const std::string& nm)
 {
   return do_clear_symbol (nm);
 }
 
 DEFCMD (clear, args, ,
   "-*- texinfo -*-\n\
-@deffn {Command} clear [-x] pattern @dots{}\n\
+@deffn {Command} clear [options] pattern @dots{}\n\
 Delete the names matching the given patterns from the symbol table.  The\n\
 pattern may contain the following special characters:\n\
 \n\
 @table @code\n\
 @item ?\n\
 Match any single character.\n\
 \n\
 @item *\n\
@@ -2425,17 +2425,34 @@ variables (local and global) are cleared
 @code{clear} is called with at least one argument, only the visible\n\
 names matching the arguments are cleared.  For example, suppose you have\n\
 defined a function @code{foo}, and then hidden it by performing the\n\
 assignment @code{foo = 2}.  Executing the command @kbd{clear foo} once\n\
 will clear the variable definition and restore the definition of\n\
 @code{foo} as a function.  Executing @kbd{clear foo} a second time will\n\
 clear the function definition.\n\
 \n\
-With -x, clear the variables that don't match the patterns.\n\
+The following options are available in both long and short form\n\
+@table @code\n\
+@item -all, -a\n\
+Clears all local and global user-defined variables and all functions\n\
+from the symbol table.\n\
+\n\
+@item -exclusive, -x\n\
+Clears the variables that don't match the following pattern.\n\
+\n\
+@item -functions, -f\n\
+Clears the function names and the built-in symbols names.\n\
+@item -global, -g\n\
+Clears the global symbol names.\n\
+@item -variables, -v\n\
+Clears the local variable names.\n\
+@end table\n\
+With the execption of @code{exclusive}, all long options can be used \n\
+without the dash as well.\n\
 @end deffn")
 {
   octave_value_list retval;
 
   int argc = args.length () + 1;
 
   string_vector argv = args.make_argv ("clear");
 
